lib/src/audio_source.dart

```diff
@@ -80,6 +83,31 @@ class AudioSource {
   /// the user can listen ie when a sound ends or key events (TODO)
   Stream<StreamSoundEvent> get soundEvents => soundEventsController.stream;
```

Comment:
That's an interesting idea. I'd say let's keep it in the pocket. I would like to avoid premature optimization that could have weird consequences or introduce unintended complexities. For example, should soloud just "eat" (i.e. not even publish) all events for sounds that aren't being listened to? Probably. But what if there's a race condition, where something happens in the same event loop and we expect to get notified, but we miss that event because we subscribe too late.

I'm not saying I have an actual, concrete issue in mind. But I've been trained to avoid optimization until there's proof that it's needed.

For now, at least I made the StreamControllers `late`. This just means they will not exist until something's either trying to listen to them OR something's trying to add an event to them. So this means it'll save us the trouble for audio sources that are never played, I guess.

Reply:
Great! Merging...